js-core/homeworks/homework-9/src/main.js

```diff
@@ -0,0 +1,207 @@
+//Task1
+let developer1 = {
+  skills: ['JavaScript', 'linux', 'html', 'OOP', 'Node.js'],
+  requirements: ['Node.js', 'JavaScript', 'OOP'],
+  goodDev: goodDev
+};
+let developer2 = {
+  experience: [
+    { technology: 'java' },
+    { technology: 'c++' },
+    { technology: 'aws' },
+    { technology: 'docker' }
+  ],
+  requirements: ['java', 'json', 'c++', 'JavaScript'],
+  goodDev: goodDev
+};
+
+function goodDev(dev) {
+	let arrRequirements = this.requirements;
+	if(this.skills) {
+		let arrSkills = this.skills;
+		arrRequirements.forEach(function(elem) {
+			let initialElem = elem;
+			let comparison = arrSkills.some(function(elem) {
+				if(typeof elem == 'object') {
+					for(let key in elem) {
+						return elem[key] == initialElem 
+					};
+				};
+				return initialElem == elem;
+			});
+			if (comparison) {
+				console.log( `required: ${elem} ...success`);
+			} else {
+				console.log( `required: ${elem} ...fail`);
+			};
+		});
+	} else if(this.experience) {
+		let arrExperience = this.experience;
+		arrRequirements.forEach(function(elem) {
+			let initialElem = elem;
+			let comparison = arrExperience.some(function(elem) {
+				if(typeof elem == 'object') {
+					for(let key in elem) {
+						return elem[key] == initialElem 
+					};
+				};
+				return initialElem == elem;
+			});
+			if (comparison) {
+				console.log( `Task1---> required: ${elem} ...success`);
+			} else {
+			console.log( `Task1---> required: ${elem} ...fail`);
+			};
+		});
+	};
+};
+
+//developer1.goodDev();
+developer2.goodDev();
+
+
+//Task2
+let myObject = {
+  database: [
+    { age: 100, name: 'b' },
+    { age: 15, name: 'c' },
+    { age: 25, name: 'a' }
+  ]
+};
+
+myObject.myFilter = function(param) {
+	let tmp = this.database.slice();
```

Comment:
instead of slice you can use destruction, it would closer to ES6 and common use-cases